pkg/app/pipectl/cmd/initialize/ecs.go

```diff
@@ -0,0 +1,90 @@
+// Copyright 2023 The PipeCD Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package initialize
+
+import (
+	"fmt"
+
+	"github.com/pipe-cd/pipecd/pkg/app/pipectl/prompt"
+	"github.com/pipe-cd/pipecd/pkg/config"
+)
+
+// Use genericConfigs in order to simplify using the GenericApplicationSpec and keep the order as we want.
+type genericECSApplicationSpec struct {
+	Name        string                    `json:"name"`
+	Input       config.ECSDeploymentInput `json:"input"`
+	Description string                    `json:"description,omitempty"`
+}
+
+func generateECSConfig(reader prompt.Reader) (*genericConfig, error) {
+	spec, e := generateECSSpec(reader)
+	if e != nil {
+		return nil, e
+	}
```

Comment:
[imo] It would be nice to define necessary values first, and pass them to `generateECSSpec`  👀  WDYT?

Reply:
Thanks, I refactored